server.js

```diff
@@ -1,30 +1,90 @@
-import express from "express";
-import cors from "cors";
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
-const port = process.env.PORT || 8080;
-const app = express();
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
-
-// Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+import express from 'express'
+import cors from 'cors'
+import expressListEndpoints from 'express-list-endpoints'
+
+import topMusicData from './data/top-music.json'
+
+const port = process.env.PORT || 8080
+const app = express()
+
+//middleware
+app.use(cors())
+app.use(express.json())
+
+//routes
+
+//showing documentation of endpoints
+app.get('/', (req, res) => {
+	const endpoints = expressListEndpoints(app)
+	res.json(endpoints)
+})
+
+//route that shows all data from jsonfile
+app.get('/songs', (req, res) => {
+	res.json(topMusicData)
+})
+
+//route to find and show song by id
+app.get('/songs/:songId', (req, res) => {
+	const { songId } = req.params
+
+	const song = topMusicData.find((song) => +songId === song.id)
+
+	if (song) {
+		res.json(song)
+	} else {
+		res.status(404).send('song not found')
+	}
+})
+
+// route to map all the artists names
+app.get('/artists', (req, res) => {
+	const artistNames = topMusicData.map((song) => song.artistName)
+
+	if (artistNames.length > 0) {
+		res.json(artistNames)
+	} else {
+		res.status(404).send('No artists found')
+	}
+})
+
+//route to sort all the songs by popylarity
+app.get('/favorites', (req, res) => {
+	const popular = topMusicData.sort((a, b) => +b.popularity - +a.popularity)
+
+	if (popular.length > 0) {
+		res.json(popular)
+	} else {
+		res.status(404).send('could not sort by popularity')
+	}
+})
+
+//route to sort by popularity and show the top ten songs by slice
+app.get('/favorites/:top10', (req, res) => {
```

Comment:
I am not entirely certain the URL should be written like this?
Because I thought the ":" signals you want to use a param and if I saw correctly when testing your endpoint there is only "top10" and no other params a user can add. So I guess the URL should be: /favorites/top10 ? 🙈 
However your code seems to work, so I could be wrong. 🤷‍♀️